### PR TITLE
VZ-10445: Automated test for OCNE cluster upgrade

### PR DESCRIPTION
**Pull request by prasad-shirodkar**
Provide a summary of the change and which issue (i.e. ticket) is fixed.
If there are any dependencies, for example on a PR in another repository, list those.

### CODE REVIEW

File: tests/e2e/clusterapi/ocne-driver/environment.go [hunk 1]
```diff
@@ -207,6 +212,20 @@ func fillOCNEMetadata(log *zap.SugaredLogger) error {
 		coreDNSFallback = contentStruct.ContainerImages.Coredns
 		etcdFallback = contentStruct.ContainerImages.Etcd
 		tigeraFallback = contentStruct.ContainerImages.TigeraOperator
+
+		k8sSemVer, err := semver.NewVersion(k8sVersion)
```
Comment (marcogelle) [inline tests/e2e/clusterapi/ocne-driver/environment.go:216]: To be consistent with the rest of the Kubernetes version code you added, perhaps `kubernetesFallback` should be assigned the value from `k8sSemVer` as well? You can use your own discretion whether you think this is a good idea or not though.

File: tests/e2e/clusterapi/ocne-driver/ocne_cluster_driver_test.go [hunk 2]
```diff
@@ -235,4 +241,72 @@ var _ = t.Describe("OCNE Cluster Driver", Label("f:rancher-capi:ocne-cluster-dri
...
+		})
+
+		// Update the cluster
+		t.It("update OCNE cluster with the maximum OCNE supported Kubernetes version and related info", func() {
+			if !ocneMetadataItemToInstall.KubernetesVersion.LessThan(ocneMetadataItemToUpgrade.KubernetesVersion) {
+				Skip(skipOCNEUpgradeMessage)
+			}
+			Eventually(func() error {
+				mutateFn := func(config *RancherOCNECluster) {
+					// setting an invalid kubernetes version
```
Comment (marcogelle) [inline tests/e2e/clusterapi/ocne-driver/ocne_cluster_driver_test.go:289]: This should be a valid K8s version, right? If so, remove/edit this comment.

File: tests/e2e/clusterapi/ocne-driver/environment.go (hunk 1 above)
Comment (prasad-shirodkar) [reply tests/e2e/clusterapi/ocne-driver/environment.go]: I could go either way. I renamed it.

File: tests/e2e/clusterapi/ocne-driver/ocne_cluster_driver_test.go (hunk 2 above)
Comment (prasad-shirodkar) [reply tests/e2e/clusterapi/ocne-driver/ocne_cluster_driver_test.go]: Removed comment

**Review by marcogelle — APPROVED** [review]

**Pull request merged**